### ISSUE TITLE
Where is the loss function code?

### ISSUE THREAD

**Issue by emadboctorx** [AlexeyAB/darknet#5801]
Can someone show me where is the code for the loss function?

**Comment by emadboctorx** [comment]
Is this yolo for object detection repository that's supposed to create a neural network model that has some loss function that determines the loss value during the training? or am i mistaken and this repository does something different? Excuse my ignorance as I come from a python background and i'm assuming that neural networks would work the same in python and c++ if this statement evaluates to true, please indicate where is the loss function, it's not a damn hard question to answer!

**Comment by AlexeyAB** [comment]
Because there is no automatic differentiation and automatic execution of chain rules, then all the gradients have to be implemented manually in C: 
https://github.com/AlexeyAB/darknet/blob/c625cc6503e9ccac41ecdd352a90fdff6d2a7859/src/yolo_layer.c#L156-L308

**Comment by emadboctorx** [comment]
Thank you Alexey, I was about to go insane over the matter, anyway by now I have a working implementation in tensorflow 2.2 that supports loading from your .cfg, files(tested on yolov3 and about to test it on yolov4) which you can check [here](https://github.com/emadboctorx/yolov3-keras-tf2) I'd really appreciate a feedback from you and thanks again.
